File: Java/2020/Junior/Q2.java
package Junior;

import java.util.*;

public class Q2 {
    public static void main(String[] args) {
	
        Scanner input = new Scanner(System.in);
	    
	    int P = input.nextInt(); // when to stop the program after it reaches a certain number of people
	    int N = input.nextInt(); // number of people who have the virus on day 0
	    int R = input.nextInt(); // number of people infected each succeeding day
	    
	    int days = 0;
	    int infected = N;
	    int infected_total = N;
	    
	    while (infected_total < P) {
	    	infected = infected * R;
	        infected_total += infected;
	        days=days+1;
	    }
	    
        System.out.println(days);
	    input.close();
	}

}

File: Python/2020/Junior/Q2.py
#Variables
P = int(input())#when we want the program to notify us after P number of people have been infected
N = int(input())#the number of people who have the disease on Day 0
R = int(input())#the number of people spread each day

infected = N #number of people infected at that instance
infected_total = N #total number of people infected
days = 0 #the number of days passed

for i in range (1000000000):
    if infected_total > P:
        print(days)
        break
    
    infected = infected * R
    infected_total += infected
    days+=1